optaplanner-core/src/test/java/org/optaplanner/core/api/solver/SolverManagerTest.java

```diff
@@ -240,4 +255,170 @@ public void skipAhead() throws ExecutionException, InterruptedException {
         assertTrue(eventCount.get() < 4);
     }
 
+    @Ignore("A test that runs forever and is here to investigate a faulty thread-creating behavior.")
+    @Test
+    public void solveMultipleThreadedMovesWithSolverManager_allGetSolved() throws ExecutionException, InterruptedException {
```

Comment:
@michaltomco would you please create a Jira for making it a runnable test?

Reply:
Done: https://issues.redhat.com/browse/PLANNER-1833